MobilityData/mobility-feed-api
# feat: Form submission authentication

closes #556 
**Summary:**

Users will not be able to submit forms unless they are logged in.

**Expected behavior:** 

1. If the user is not logged in, they will not be able to submit the form. They will be prompted to login
2. If the user selects the option to login through the submission page, once successfully logged in, they will return to the submission page. It will also show an alert saying "You logged in, now you can submit"

---

I set a query param when navigating from the "Add feed form" to indicate if successful to navigate back to the "Add feed form" page. On the way back I added a `from` state to the router to display a "Successfully logged in" Alert

**Testing tips:**

1. Go to `/contribute` (Add feed) while not logged in. You should see elements prompting you to sign up or login
2. Login with any method, or sign up. After that, you should be brought back to the "Add feed" page

Please make sure these boxes are checked before submitting your pull request - thanks!


Unauthenticated
![Screenshot 2024-07-31 at 08 11 55](https://github.com/user-attachments/assets/28d893f1-0d2e-4f0a-b413-f6e81a9bb65a)

Authenticated
![Screenshot 2024-07-31 at 08 11 43](https://github.com/user-attachments/assets/41974e07-25fb-44c6-a31b-1badaed3ca2d)
